pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -83,6 +84,10 @@ type watcher struct {
 	// Maximum timestamp of the last Event read.
 	// A map from repo-id to the UNIX timestamp that has been read.
 	milestoneMap sync.Map
+	// Cache for the last scanned commit for each application.
+	lastScannedCommits sync.Map
+	// Cache for the last scanned the event watcher config in application.
+	lastScannedEventWatcherConfig sync.Map
```

Comment:
How about defining a new internal struct to hold hash value and config?
Then we just need a single map from appID to that struct here.

Reply:
Thank you for your review and that's a good idea! so, I applied it by https://github.com/pipe-cd/pipecd/pull/3729/commits/5a0f982faa4be218909b51983d0473dd30150b99